neqo-transport/src/recovery.rs

```diff
@@ -533,6 +588,96 @@ impl LossRecovery {
             .min_by_key(|&(_, time)| time)
             .map(|(spc, val)| (spc, val + self.loss_delay()))
     }
+
+    /// Find when the last ack eliciting packet was sent.
+    pub fn get_min_pto(&self) -> Option<Instant> {
+        // TODO ignore PNSpace::Application until handshake is done -> a server side problem.
+        PNSpace::iter()
+            .filter_map(|spc| {
+                if self.spaces[*spc].ack_eliciting_outstanding() {
+                    // None is ordered less than Some(_). Bad. Filter them out.
+                    if let Some(time) = self.spaces[*spc].time_of_last_sent_ack_eliciting_packet {
+                        Some(time + self.pto_timeout(*spc))
+                    } else {
+                        debug_assert!(false);
+                        None
+                    }
+                } else {
+                    None
+                }
+            })
+            .min_by_key(|&time| time)
+    }
+
+    pub fn get_min_pto_pn_space(&self, now: Instant) -> Option<PNSpace> {
+        PNSpace::iter()
+            .filter_map(|spc| {
+                if self.spaces[*spc].ack_eliciting_outstanding() {
+                    if let Some(time) = self.spaces[*spc].time_of_last_sent_ack_eliciting_packet {
+                        if time <= (now - self.pto_timeout(*spc)) {
```

Comment:
I would use `time + self.pto_timeout(*spc) <= now` to avoid weird underflow bugs.  Instant is weird.